solidity/contracts/libs/isms/MultisigIsmMetadata.sol

```diff
@@ -88,4 +67,65 @@ library MultisigIsmMetadata {
         uint256 _end = _start + SIGNATURE_LENGTH;
         return _metadata[_start:_end];
     }
+
+    function suffix(
+        bytes calldata _metadata,
+        uint256 offset,
+        uint256 size
+    ) private pure returns (bytes calldata) {
+        uint256 _end = _metadata.length - offset;
+        uint256 _start = _end - size;
+        return _metadata[_start:_end];
+    }
+
+    function suffixType(bytes calldata _metadata)
+        internal
+        pure
+        returns (SuffixType)
+    {
+        return
+            SuffixType(uint8(bytes1(suffix(_metadata, SUFFIX_TYPE_OFFSET, 1))));
+    }
+
+    /**
+     * @notice Returns the merkle root of the signed checkpoint.
+     * @param _metadata ABI encoded Multisig ISM metadata.
+     * @return Merkle root of the signed checkpoint
+     */
+    function root(bytes calldata _metadata) internal pure returns (bytes32) {
+        assert(suffixType(_metadata) == SuffixType.ROOT);
```

Comment:
wdyt of not having these asserts and just documenting that certain functions should only be used depending on the type? Just noting that it's using gas sorta unnecessarily

Reply:
its 1 byte of calldata so imo the assertions are worth it